src/Compilers/VisualBasic/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.vb

```diff
@@ -0,0 +1,834 @@
+                model.GetSymbolInfo(invocation).CandidateSymbols.Select(Function(c) c.ToTestDisplayString()))
+        End Sub
+
+        <Fact>
+        Public Sub TestNamedParams()
+            Dim source =
+<compilation>
+    <file name="Program.vb">
+Class C
+    Shared Sub M(ByVal ParamArray c() As Integer)
+    End Sub
+    Shared Sub Main()
+        M(c:=1, 2)
+    End Sub
+End Class
+    </file>
+</compilation>
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=latestParseOptions)
+
+            comp.AssertTheseDiagnostics(<errors>
+BC30587: Named argument cannot match a ParamArray parameter.
+        M(c:=1, 2)
+          ~
+                                        </errors>)
+
+            Dim tree = comp.SyntaxTrees.First()
+            Dim model = comp.GetSemanticModel(tree)
+            Dim nodes = tree.GetCompilationUnitRoot().DescendantNodes()
+            Dim invocation = nodes.OfType(Of InvocationExpressionSyntax)().Single()
+            Assert.Equal("M(c:=1, 2)", invocation.ToString())
+            AssertEx.Equal({"Sub C.M(ParamArray c As System.Int32())"},
+                model.GetSymbolInfo(invocation).CandidateSymbols.Select(Function(c) c.ToTestDisplayString()))
+        End Sub
+
+        <Fact>
+        Public Sub TestNamedParams2()
+
+            Dim source =
+<compilation>
+    <file name="Program.vb">
+Class C
+    Shared Sub M(ParamArray x() As Integer)
+    End Sub
+    Shared Sub Main()
+        M(1, x:=2)
+    End Sub
+End Class
+    </file>
+</compilation>
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=latestParseOptions)
+            comp.AssertTheseDiagnostics(<errors>
+BC30587: Named argument cannot match a ParamArray parameter.
+        M(1, x:=2)
+             ~
+                                        </errors>)
+
+            Dim tree = comp.SyntaxTrees.First()
+            Dim model = comp.GetSemanticModel(tree)
+            Dim nodes = tree.GetCompilationUnitRoot().DescendantNodes()
+            Dim invocation = nodes.OfType(Of InvocationExpressionSyntax)().Single()
+            Assert.Equal("M(1, x:=2)", invocation.ToString())
+            Assert.Null(model.GetSymbolInfo(invocation).Symbol)
+        End Sub
+
+        <Fact>
+        Public Sub TestTwiceNamedParams()
+
+            Dim source =
+<compilation>
+    <file name="Program.vb">
+Class C
+    Shared Sub M(ParamArray x() As Integer)
+    End Sub
+    Shared Sub Main()
+        M(x:=1, x:=2)
+    End Sub
+End Class
+    </file>
+</compilation>
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=latestParseOptions)
+            comp.AssertTheseDiagnostics(<errors>
+BC30587: Named argument cannot match a ParamArray parameter.
+        M(x:=1, x:=2)
+          ~
+BC30587: Named argument cannot match a ParamArray parameter.
+        M(x:=1, x:=2)
+                ~
+                                        </errors>)
```

Comment:
Consider also testing the a twice named parameter that is not `ParamArray` since VB does not allow naming that parameter. In short, add a version of the test below, with 15.6.